### PR TITLE
fdttools: add fdt utility tools

### PR DESCRIPTION
**Pull request by leo11261**
## Summary
add fdt utility tools, work with https://github.com/apache/nuttx/pull/10121

## Impact
tools for fdt

## Testing
internal testing

### CODE REVIEW

**Comment by btashton** [comment]
With the fdt procfs support in you should be able to demonstrate dumping that from a qemu build. I don't think "internal testing" is sufficient evidence that this works. It's fine that there is not an in tree config yet.

File: system/fdt/Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  MAINSRC += fdtoverlay.c
+  PROGNAME += fdtoverlay
+  STACKSIZE += $(CONFIG_SYSTEM_FDTOVERLAY_STACKSIZE)
+  PRIORITY += $(CONFIG_SYSTEM_FDTOVERLAY_PRIORITY)
+endif
+
+CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/system/fdt/
+CFLAGS += ${INCDIR_PREFIX}$(TOPDIR)/libs/libc/dtc/libfdt/
+
+VPATH += ${INCDIR_PREFIX}$(TOPDIR)/libs/libc/dtc/
```
Comment (anchao) [inline system/fdt/Makefile:58]: VPATH += $(TOPDIR)/libs/libc/dtc/

File: system/fdt/Makefile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+DEPPATH += --dep-path ${INCDIR_PREFIX}$(TOPDIR)/libs/libc/dtc/
```
Comment (anchao) [inline system/fdt/Makefile:59]: DEPPATH += --dep-path $(TOPDIR)/libs/libc/dtc/

File: system/fdt/version_gen.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __SYSTEM_FDT_VERSION_GEN_H
+#define __SYSTEM_FDT_VERSION_GEN_H
+
+/* DTC Version should be same as that in nuttx/libs/libc/dtc/ */
+
+#define DTC_VERSION "DTC 1.7.0"
```
Comment (anchao) [inline system/fdt/version_gen.h:26]: Add a fdt config(CONFIG_LIBFDT_VERSION) to control the version

**Pull request merged**